operations/model.go

```diff
@@ -253,6 +257,24 @@ func (s *ClientSettings) getApiServerHost(useCorp bool) string {
 	return s.APIServerHost
 }
 
+func checkCLIVersion(c client.Communicator) error {
+	clients, err := c.GetClientConfig(context.Background())
+	if err != nil {
+		grip.Debug(errors.Wrap(err, "getting client config info"))
+	}
+	if clients != nil && clients.OldestAllowedCLIVersion != "" {
+		isCLIVersionTooOld, err := isFirstDateBefore(evergreen.ClientVersion, clients.OldestAllowedCLIVersion)
+		if err != nil {
+			grip.Warning(errors.Wrap(err, "checking if client is older than the latest version"))
+		}
+		if isCLIVersionTooOld {
+			return errors.Errorf("CLI version '%s' is older than the oldest allowed CLI version '%s'. "+
```

Comment:
This has the potential to break a lot of people's setup. I'd suggest sending out some user communications before deploying this so they know to expect it, understand why we chose to do it, and understand how to prepare for it.